File: android-app/src/main/java/com/medziku/plantwatering/data/parts/ThermometerImpl.java
package com.medziku.plantwatering.data.parts;

import android.os.Parcel;
import android.os.Parcelable;

import com.medziku.plantwatering.station.parts.Thermometer;


public class ThermometerImpl extends Thermometer implements Parcelable {

    public ThermometerImpl(String name, float temperature) {
        super(name, temperature);
    }

    private ThermometerImpl(Parcel in) {
        super(in.readString(), in.readFloat());
    }

    public static final Creator<ThermometerImpl> CREATOR = new Creator<ThermometerImpl>() {
        @Override
        public ThermometerImpl createFromParcel(Parcel in) {
            return new ThermometerImpl(in);
        }

        @Override
        public ThermometerImpl[] newArray(int size) {
            return new ThermometerImpl[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(getName());
        dest.writeFloat(getTemperature());
    }
}

File: android-app/src/main/kotlin/com/medziku/plantwatering/data/parts/ThermometerImpl.kt
package com.medziku.plantwatering.data.parts

import android.os.Parcel
import android.os.Parcelable

import com.medziku.plantwatering.station.parts.Thermometer


class ThermometerImpl : Thermometer, Parcelable {

    constructor(name: String, temperature: Float) : super(name, temperature)

    private constructor(`in`: Parcel) : super(`in`.readString(), `in`.readFloat())

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(name)
        dest.writeFloat(temperature)
    }

    companion object {

        @JvmField
        val CREATOR: Parcelable.Creator<ThermometerImpl> = object : Parcelable.Creator<ThermometerImpl> {
            override fun createFromParcel(`in`: Parcel): ThermometerImpl {
                return ThermometerImpl(`in`)
            }

            override fun newArray(size: Int): Array<ThermometerImpl?> {
                return arrayOfNulls(size)
            }
        }
    }
}
